### ISSUE TITLE
False positive in force_unwrapping when returning collection from function

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2042]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "func foo() -> [AnyHashable: Any]\!" | swiftlint lint --no-cache --use-stdin --enable-all-rules 
Loading configuration from '.swiftlint.yml'
<nopath>:1:33: warning: Force Unwrapping Violation: Force unwrapping should be avoided. (force_unwrapping)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.24.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - force_unwrapping
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. No
* Which Xcode version are you using (check `xcode-select -p`)? 9.2
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
func foo() -> [AnyHashable: Any]!
func bar() -> [Int]!
```

**Comment by Coeur** [comment]
If the rule `force_unwrapping` doesn't trigger a violation for those, then I can declare:

```
extension Optional {
    var forceUnwrapped: Wrapped! {
        return self
    }
}
```

Then I replace all the `foo!` with `foo.forceUnwrapped` and SwiftLint will never trigger the `force_unwrapping` rule.
